verifiers/internal/gha/slsaprovenance/v1.0/byob.go

```diff
@@ -0,0 +1,245 @@
+package v1
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	intoto "github.com/in-toto/in-toto-golang/in_toto"
+	slsa1 "github.com/in-toto/in-toto-golang/in_toto/slsa_provenance/v1"
+	serrors "github.com/slsa-framework/slsa-verifier/v2/errors"
+
+	"github.com/slsa-framework/slsa-verifier/v2/verifiers/internal/gha/slsaprovenance/common"
+)
+
+var (
+	// byobBuildType is the base build type for BYOB delegated builders.
+	byobBuildType = "https://github.com/slsa-framework/slsa-github-generator/delegator-generic@v0"
+
+	// containerBasedBuildType is the build type for the container-based builder and is based on BYOB.
+	containerBasedBuildType = "https://slsa.dev/container-based-build/v0.1?draft"
```

Comment:
note: the container-based builder is not based no BYOB, iiuc https://github.com/slsa-framework/slsa-github-generator/blob/main/.github/workflows/builder_container-based_slsa3.yml

Reply:
Right, I'll refactor this a bit.